File: src/main/java/inno/jago/ast/SourceFileNode.java
package inno.jago.ast;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
public class SourceFileNode extends Entity {

    @Getter
    @Setter
    private PackageNode packageNode;

//    private List<ImportNode>

//    private List<r>

//    packageClause ( END ) ( importDecl ( END ) )* ( topLevelDecl ( END ) )* EOF;
}

File: src/main/kotlin/inno/jago/ast/SourceFileNode.kt
package inno.jago.ast

import inno.jago.ast.decl.TypeDeclarationNode
import inno.jago.entity.Entity
import inno.jago.lexer.Pos

class SourceFileNode(
    pos: Pos,
    val packageName: PackageNode,
    val importNodes: List<ImportNode>,
    val topLevelDecls: List<TypeDeclarationNode>
) : Entity(pos = pos)
